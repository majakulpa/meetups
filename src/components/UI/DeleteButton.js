import React from 'react'
import Swal from 'sweetalert2'
import { HiOutlineTrash } from 'react-icons/hi'

const deleteButton = ({
  text,
  alertText,
  handleDelete,
  confirmText,
  textFireDeleted,
  textFireSafe
}) => {
  return (
    <button
      onClick={e =>
        Swal.fire({
          title: 'Are you sure?',
          text: alertText,
          icon: 'warning',
          showCancelButton: true,
          confirmButtonColor: '#805ad5',
          cancelButtonColor: '#a0aec0',
          confirmButtonText: confirmText,
          reverseButtons: true
        }).then(result => {
          if (result.value) {
            e.persist()
            handleDelete(e)
            Swal.fire('Deleted!', textFireDeleted, 'success')
          } else if (result.dismiss === Swal.DismissReason.cancel) {
            Swal.fire('Cancelled', textFireSafe, 'error')
          }
        })
      }
      className="block bg-gray-500 float-right hover:bg-gray-600 text-white tracking-wide flex
          capitalize py-2 px-4 rounded focus:bg-gray-600 focus:outline-none focus:shadow-outline"
    >
      <HiOutlineTrash className="mt-1 mr-1 font-bold" />
      <span>{text}</span>
    </button>
  )
}

export default deleteButton
