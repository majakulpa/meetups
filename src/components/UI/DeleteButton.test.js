import React from 'react'
import ReactDOM from 'react-dom'
import '@testing-library/jest-dom/extend-expect'
import { cleanup } from '@testing-library/react'
import DeleteButton from './DeleteButton'
import renderer from 'react-test-renderer'

afterEach(cleanup)

test('if renders', () => {
  const div = document.createElement('div')
  ReactDOM.render(<DeleteButton />, div)
})

test('if matches snapshot', () => {
  const tree = renderer.create(<DeleteButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
