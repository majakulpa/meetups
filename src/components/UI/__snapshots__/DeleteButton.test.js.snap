// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<button
  className="block bg-gray-500 float-right hover:bg-gray-600 text-white tracking-wide flex capitalize py-2 px-4 rounded focus:bg-gray-600 focus:outline-none focus:shadow-outline"
  onClick={[Function]}
>
  <svg
    className="hidden sm:block mt-1 mr-1 font-bold"
    fill="none"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
  <span
    className="text-xs sm:text-base"
  />
</button>
`;
